src/main/scala/higherkindness/rules_scala/workers/common/AnnexMapper.scala

```
@@ -38,10 +38,15 @@ object AnnexMapper {
    */
   final def getConsistentWriteStamp(stamp: Stamp): Stamp = {
     stamp match {
-      case farmHash: FarmHash         => farmHash
-      case hash: Hash                 => hash
+      case farmHash: FarmHash => farmHash
+      case hash: Hash => hash
       case lastModified: LastModified => new LastModified(JarHelper.DEFAULT_TIMESTAMP)
-      case _                          => throw new Exception("Unexpected Stamp type encountered when writing.")
+      case stamp => {
+        println(
```

Logging the stamp seems like a good idea, but won't we still want to throw an exception here?
Otherwise, we might get nondeterminism, right?

---

:man_shrugging: I mean, I have no idea. I only have the vaguest notion of what a stamp is. Are you thinking this is less a bug of not handling other stamp types and more of us trying to use stamp types that are non-deterministic?

---

The exception is being thrown because we're not handling a certain stamp type that's being given to us. I too am not sure exactly what this function is doing; I think you'll have to read the code to get a sense for what's calling this and what stamp types may be provided to us.

I just know that the purpose of it is to eliminate non-determinism issues by converting nondeterministic stamps (e.g. timestamps) to deterministic ones, so silently succeeding doesn't seem like the move here. But getting more visibility would be valuable.